File: backend/src/main/java/com/app/backend/domain/category/repository/CategoryRepository.java
package com.app.backend.domain.category.repository;

import com.app.backend.domain.category.entity.Category;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface CategoryRepository extends JpaRepository<Category, Long> {
    boolean existsByName(String name);

    Category findByName(String name);

    Page<Category> findByDisabledFalse(Pageable pageable);

    Optional<Category> findByNameAndDisabled(String name, Boolean disabled);
}

File: backend/src/main/java/com/app/backend/domain/category/repository/CategoryRepository.kt
package com.app.backend.domain.category.repository

import com.app.backend.domain.category.entity.Category
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface CategoryRepository : JpaRepository<Category, Long> {

    fun existsByName(name: String): Boolean

    fun findByName(name: String): Category?

    fun findByDisabledFalse(pageable: Pageable): Page<Category>

    fun findByNameAndDisabled(name: String, disabled: Boolean): Optional<Category>
}
